### PR TITLE
feat: add icon to collapsible

### LINKED ISSUE #37379: collapse折叠点击区域是否在header和disable基础上加个icon
### What problem does this feature solve?

有点击icon进行折叠展开的需求，但现在只有两个属性值，不想点击header文字来进行展开折叠

### What does the proposed API look like?

希望增加一个点击折叠icon进行展开折叠， 点击文字不进行折叠

### PR DESCRIPTION
**Pull request by MattinaYang**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

[https://github.com/react-component/collapse/pull/232](https://github.com/react-component/collapse/pull/232)
close #37379

### 💡 Background and solution

I added 'icon' to rc-collapse's 'collapsible' segment several monthes ago, and the pr had been merged two days ago, since version 3.4.0 was released. So I wanna add this to antd Collapse too.

### 📝 Changelog

Add choice 'icon' to 'collapsible', which could only be 'header' or 'disabled' for now.

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Collapse supports `collapsible="icon"` to collapse by clicking icon. |
| 🇨🇳 Chinese | Collapse 新增 `collapsible="icon"` 从而支持点击图标展开收起。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/37419#issuecomment-1238812797 这个问题没解，这个需要改 rc-collapse

**Comment by Sheepeer** [comment]
> [#37419 (comment)](https://github.com/ant-design/ant-design/pull/37419#issuecomment-1238812797) 问题没解，这个需要改 rc-collapse

这个问题在 feature 分支（rc-collapse为3.3.0版本）就出现了，似乎不是添加‘icon’选项带来的？ 👀

**Comment by afc163** [comment]
https://github.com/react-component/collapse/pull/238 这里引入的，3.4.0 里的问题。

**Comment by afc163** [comment]
我来修掉。https://github.com/react-component/collapse/pull/240

File: package.json [hunk 1]
```diff
@@ -126,7 +126,7 @@
     "moment": "^2.29.2",
     "rc-cascader": "~3.6.0",
     "rc-checkbox": "~2.3.0",
-    "rc-collapse": "~3.3.0",
+    "rc-collapse": "~3.4.0",
```
Comment (afc163) [inline package.json:129]:
```suggestion
    "rc-collapse": "~3.4.1",
```

**Comment by afc163** [comment]
现在出现了一个新的问题，@Sheepeer 看看

https://github.com/ant-design/ant-design/actions/runs/3072589779/jobs/4964204830

File: components/collapse/demo/collapsible.md [hunk 2]
```diff
@@ -32,6 +32,11 @@ const App: React.FC = () => (
         <p>{text}</p>
       </Panel>
     </Collapse>
+    <Collapse collapsible="icon" defaultActiveKey={['1']}>
+      <Panel header="This panel can only be collapsed by clicking icon" key="1">
+        <p>{text}</p>
+      </Panel>
+    </Collapse>
```
Comment (afc163) [inline components/collapse/demo/collapsible.md:39]: 这里需要更新一下 jest snapshot。

Comment (Sheepeer) [reply components/collapse/demo/collapsible.md:39]: 已更新

File: components/collapse/index.zh-CN.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/IxH16B9RD/Collapse.svg
 | accordion | 手风琴模式 | boolean | false |  |
 | activeKey | 当前激活 tab 面板的 key | string\[] \| string <br/> number\[] \| number | 默认无，accordion 模式下默认第一个元素 |  |
 | bordered | 带边框风格的折叠面板 | boolean | true |  |
-| collapsible | 所有子面板是否可折叠或指定可折叠触发区域 | `header` \| `disabled` | - | 4.9.0 |
+| collapsible | 所有子面板是否可折叠或指定可折叠触发区域 | `header` \| `icon` \| `disabled` | - | 4.9.0 |
```
Comment (afc163) [inline components/collapse/index.zh-CN.md:26]: 后面的版本要标记一下，icon 在 4.24.0 支持

File: components/collapse/index.en-US.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ A content area which can be collapsed and expanded.
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
-| collapsible | Specify whether the panel be collapsible or the trigger area of collapsible | `header` \| `icon` \| `disabled` | - | 4.9.0 |
+| collapsible | Specify whether the panel be collapsible or the trigger area of collapsible | `header` \| `icon` \| `disabled` | - | 4.24.0 |
```
Comment (afc163) [inline components/collapse/index.en-US.md:37]: 参考 https://ant.design/components/table-cn/#Table 

sticky 属性

**Comment by afc163** [comment]
图片

有点问题，尝试重装本地 node_modules 以后再跑 `npm test -- components/collapse`

**Comment by Sheepeer** [comment]
> 图片
> 
> 有点问题，尝试重装本地 node_modules 以后再跑 `npm test -- components/collapse`

ohhhhhhh真的是依赖的问题 谢谢！！！

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
图片

@zombieJ 这个 PR 貌似不会上传 argo-ci

**Comment by zombieJ** [comment]
> 图片
> 
> @zombieJ 这个 PR 貌似不会上传 argo-ci

workflow run 不知道为什么拿不到 archive 了，这个之前尝试找了一下还是不行。暂时没有头绪。

**Pull request merged**